packages/perennial/contracts/multiinvoker/MultiInvokerRollup.sol

```diff
@@ -76,80 +80,112 @@ contract MultiInvokerRollup is IMultiInvokerRollup, MultiInvoker {
      * @param input Packed data to pass to invoke logic
      */
     function _decodeFallbackAndInvoke(bytes calldata input) internal {
-        PTR memory ptr;
+        PTR memory ptr = PTR({pos: 1}); // INVOKE_ID must be first byte to prevent collisions
```

Comment:
it might be good to check that the first byte matches the magic byte to prevent someone from forgetting to include it and not tripping another function.

Reply:
we could also do 

```
PTR memory ptr;
if(input[0:1] != 0x45) {
  ptr.pos = 0;
} else {
  ptr.pos == 1
}
```
if we want to make it optional, otherwise we can do:
```
 fallback (bytes calldata input) external returns (bytes memory) {
        if(input[0:1] != 0x45) revert("must prepend magic byte 0x45. prevents sig collisions");
        _decodeFallbackAndInvoke(input);
        return "";
    }
```

... which seems restricted but the revert message is pretty clear

Reply:
let's do the second one, but use custom errors so something like `revert MultiInvokerRollupMIssingMagicByteError()`